packages/react/src/checkout/draftOrders/hooks/useDraftOrders.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  areDraftOrdersFetched,
+  areDraftOrdersLoading,
+  createDraftOrder,
+  fetchDraftOrder as fetchDraftOrderAction,
+  fetchDraftOrders,
+  getCreateDraftOrderError,
+  getDraftOrders,
+  getDraftOrdersError,
+  isCreatingDraftOrder,
+  removeDraftOrder,
+  removeDraftOrderItem,
+  resetDraftOrders,
+  type StoreState,
+  updateDraftOrder,
+  updateDraftOrderItem,
+} from '@farfetch/blackout-redux';
+import { useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import useAction from '../../../helpers/useAction.js';
+import type { DraftOrdersQuery } from '@farfetch/blackout-client';
+import type { UseDraftOrdersOptions } from './types/index.js';
+
+const useDraftOrders = (
+  query: DraftOrdersQuery,
+  options?: UseDraftOrdersOptions,
+) => {
+  const { enableAutoFetch = true, fetchConfig, orderId } = options || {};
+
+  const fetch = useAction(fetchDraftOrders);
+  const fetchDraftOrder = useAction(fetchDraftOrderAction);
+  const createAction = useAction(createDraftOrder);
+  const removeDraftOrderItemAction = useAction(removeDraftOrderItem);
+  const removeDraftOrderAction = useAction(removeDraftOrder);
+  const updateDraftOrderItemAction = useAction(updateDraftOrderItem);
+  const updateDraftOrderAction = useAction(updateDraftOrder);
+  const resetAction = useAction(resetDraftOrders);
+
+  const draftOrders = useSelector((state: StoreState) =>
+    getDraftOrders(state, query),
+  );
+
+  const error = useSelector((state: StoreState) =>
+    getDraftOrdersError(state, query),
+  );
+  const isLoading = useSelector((state: StoreState) =>
+    areDraftOrdersLoading(state, query),
+  );
+
+  const createError = useSelector((state: StoreState) =>
+    orderId ? getCreateDraftOrderError(state, orderId) : null,
+  );
+  const createIsLoading = useSelector((state: StoreState) =>
+    orderId ? isCreatingDraftOrder(state, orderId) : false,
+  );
+
+  const isFetched = useSelector((state: StoreState) =>
+    areDraftOrdersFetched(state, query),
+  );
+
+  useEffect(() => {
+    if (!isFetched && !isLoading && query && enableAutoFetch) {
+      fetch(query, fetchConfig);
+    }
+  }, [enableAutoFetch, fetch, fetchConfig, isFetched, isLoading, query]);
+
+  return {
+    isFetched,
+    isLoading,
+    createIsLoading,
+    error,
+    createError,
+    draftOrders,
+    actions: {
+      fetch,
+      fetchDraftOrder,
+      create: createAction,
+      remove: removeDraftOrderAction,
+      removeItem: removeDraftOrderItemAction,
+      update: updateDraftOrderAction,
+      updateItem: updateDraftOrderItemAction,
```

Comment:
These actions are also returned in the `useDraftOrder` hook. There can be an argument to them here as well, but by using this hook we lose access to the `error` and `loading` states for these actions. Is this deliberate? What is the use case for having them here as well?

Reply:
I think @anapim if you want to use those actions you should import the `useDraftOrder` that uses `useDraftOrders` inside.
I think is exported here to be able to do this https://github.com/Farfetch/blackout/pull/959/files#diff-35c43864f25ead47fc01471d99d5de9848e1a5f4e8edd3b97cb391fdc9d6b2ceR37

But it is a little confusing

Reply:
The difference between useDraftOrder and useDraftOrders is that useDraftOrder's actions are scoped to the passed draftOrderId and query parameters to the hook and will return the state of those requests.
useDraftOrders define the actions that are used in useDraftOrder as a convenience and are not scoped to a particular draftOrderId.

Reply:
This was thought up following the pattern of the project architecture, you can look at `useBag` or `useOrder`. However, the idea here is for the developer to have as much freedom as possible with the actions in the hooks. 

As a simple example, suppose the user wants to create a draf order, then list all the draft orders and finally update a draft order, they can do all this by importing the useDraftOrders hook.